runtime/Go/antlr/parser_rule_context.go

```diff
@@ -181,8 +193,8 @@ func (prc *BaseParserRuleContext) GetRuleContext() RuleContext {
 	return prc
 }
 
-func (prc *BaseParserRuleContext) Accept(visitor ParseTreeVisitor) interface{} {
-	return visitor.VisitChildren(prc)
+func (prc *BaseParserRuleContext) Visit(delegate ParseTreeVisitor, args ...interface{}) (result interface{}) {
+	return delegate.VisitChildren(prc, delegate)
 }
```

Comment:
So...IIRC, this is the complicated part.  Because Go doesn't actually have what you'd typically call 'inheritance', the `VisitChildren` method implemented in the `ParseTreeVisitor` will actually NOT call the right method if you just use the `*BaseParseTreeVisitor` pointer.  So IIRC he passes in `delegate` so that it can call the right methods on it.  I know that was necessary for my initial version, but I did it very differently, and without a lot of the Go experience that you and he bring to the table.  Using type assertions and type `switch` appears to have gone a long way.

I did a more 1:1 implementation of `VisitChildren`, using the Java code as the guideline, but I had to put it into the generated code in order for it to work and did a lot of ugly `if` checks.  He found a way to put it into the base code, and have it still call the right methods.  (Although ignoring default value, and a few other niceties.)

A lot of this is from memory, refreshed by looking at the various pieces; it's been a while, and I admit, I don't really understand the reason for `args`, for example. :-/